### ISSUE TITLE
[Feature request] Dynamically update layouts after resizing windows

### ISSUE THREAD

**Issue by mbi** [domferr/tilingshell#121]
**Describe the feature you'd like**

1. I apply a given layout to three windows
2. After some time, I resize some windows (and the other windows' size adapt, great!) 
3. I then drag one window out of the layout, or one of the windows resizes on its own. It now floats above the two others. 
4. If I re-apply the layout position of the "floating" window, it will snap back into the original layout's position, but since I've resized some windows earlier, it won't "fit" and will overlap the two other windows. The only solution in this case is to re-apply the layout to all three windows.

It would be nice if there was a option that when I resize the windows in step 2, the stored layout (or the layout of this "session"?) is updated, so that when I drag a window into the layout again at step 4, it'll just fit back into the current group.

**Comment by emko** [comment]
yes this would be amazing, making windows float if you drag them is nice but once i  put it back it should resize to the available space not the tiling layout old width/height

**Comment by TimYorke** [comment]
I agree - it's my main frustration. 

When you move a boundary by dragging a window edge, the current tile layout should adapt to it (but only temporarily for this instance - i.e. the saved layout shouldn't be modified unless explicitly requested).
